database/src/main/resources/db/migration/R__list_taxi_pricing_statistics.sql

```diff
@@ -12,8 +15,73 @@ BEGIN
                    FROM taxi_service_prices
                   ORDER BY service_id,
                            timestamp DESC) AS s
-          ORDER BY s.' || QUOTE_IDENT(ordering_column) || ' ' ||
-                         (CASE WHEN ordering_direction = TRUE THEN 'ASC' ELSE 'DESC' END);
+          ' || (CASE WHEN ordering_column IS NOT NULL AND ordering_direction IS NOT NULL
+                     THEN 'ORDER BY s.' || QUOTE_IDENT(ordering_column) || ' ' ||
+                          (CASE WHEN ordering_direction = TRUE
+                                THEN 'ASC'
+                                ELSE 'DESC'
+                           END)
+                     ELSE ''
+                END);
+END
+$func$
+LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION list_taxi_statistics(
+    primary_ordering_column TEXT,
+    primary_ordering_direction BOOLEAN,
+    secondary_ordering_column TEXT,
+    secondary_ordering_direction BOOLEAN)
+    RETURNS TABLE (
+                      "operator-id" integer,
+                      "service-id" integer,
+                      "name" text,
+                      "timestamp" timestamp,
+                      "start-price-daytime" numeric,
+                      "start-price-nighttime" numeric,
+                      "start-price-weekend" numeric,
+                      "price-per-minute" numeric,
+                      "price-per-kilometer" numeric,
+                      "operating-areas" text[]
+                  ) AS
+$func$
+BEGIN
+    RETURN QUERY EXECUTE '
+        SELECT o.id AS "operator-id",
+               s.id AS "service-id",
+               CONCAT(o."name", ''/'', s."name") AS "name",
+               l.timestamp AS "timestamp",
+               l.start_price_daytime AS "start-price-daytime",
+               l.start_price_nighttime AS "start-price-nighttime",
+               l.start_price_weekend AS "start-price-weekend",
+               l.price_per_minute AS "price-per-minute",
+               l.price_per_kilometer AS "price-per-kilometer",
+               (SELECT array_agg(oa_d.text)
```

Comment:
My IDEA formats `array_agg` as `ARRAY_AGG`... though PostgreSQL's own documentation uses `array_agg`.

Reply:
I suppose it wants to capitalize built-in functions to contrast them from user made functions. I'd rather not change this at this point, I need to take a step back from this horrid concatenation to get a bit better perspective on if it's even good... :)